tests/accessor/accessor_exceptions.hpp

```diff
@@ -0,0 +1,391 @@
+ *        verification functions.
+ * @tparam AccT Current type of the accessor: generic_accessor,
+ *         local_accessor, or host_accessor
+ * @tparam T Current data type
+ * @tparam AccessModeTypeT Access mode type
+ * @tparam TargetTypeT Target type
+ * @tparam DimensionTypeT Dimension type
+ * @param type_name Current data type string representation
+ * @param access_mode_name Current access mode string representation
+ * @param target_name Current target string representation
+ */
+template <typename AccT, typename T, typename AccessModeTypeT,
+          typename TargetTypeT, typename DimensionTypeT>
+class run_tests {
+  static constexpr sycl::access_mode AccessMode = AccessModeTypeT::value;
+  static constexpr int Dimension = DimensionTypeT::value;
+  static constexpr sycl::target Target = TargetTypeT::value;
+
+ public:
+  void operator()(const std::string& type_name,
+                  const std::string& access_mode_name,
+                  const std::string& target_name) {
+    constexpr accessor_tests_common::accessor_type acc_type = AccT::value;
+    if constexpr (acc_type ==
+                  accessor_tests_common::accessor_type::generic_accessor) {
+      test_exception_for_generic_acc<T, AccessMode, Dimension, Target,
+                                     acc_type>(type_name, access_mode_name,
+                                               target_name);
+    } else if constexpr (acc_type ==
+                         accessor_tests_common::accessor_type::host_accessor) {
+      test_exception_for_host_acc<T, AccessMode, Dimension, Target, acc_type>(
+          type_name, access_mode_name, target_name);
+    } else if constexpr (acc_type ==
+                         accessor_tests_common::accessor_type::local_accessor) {
+      test_exception_for_local_acc<T, Dimension, Target, acc_type>(
+          type_name, access_mode_name, target_name);
+    }
+  }
+};
+
+/**
+ * @brief Struct that runs test with different input parameters types
+ * @tparam AccT Current type of the accessor: generic_accessor,
+ *         local_accessor, or host_accessor
+ * @tparam T Current data type
+ * @param type_name Current data type string representation
+ * @param access_mode_name Current access mode string representation
+ * @param target_name Current target string representation
+ */
+template <typename T, typename AccT>
+class run_tests_with_types {
+ public:
+  void operator()(const std::string& type_name) {
+    // Type packs instances have to be const, otherwise for_all_combination
+    // will not compile
+    const auto access_modes = get_access_modes();
+    const auto dimensions = get_dimensions();
+    const auto targets = get_targets();
+    const auto cur_type =
+        named_type_pack<T>::generate(type_name_string<T>::get(type_name));
+
+    for_all_combinations<run_tests, AccT>(cur_type, access_modes, targets,
+                                          dimensions);
```

Comment:
No, by current solution we can provide a string representation of current testing data type

Reply:
Do you mean forwarding the `type_name` parameter?

The `for_all_combinations` have an ability to recognize additional template parameters and additional runtime parameters to the functor called. We are free to avoid `cur_type` construction by passing `T` as additional template parameter and `type_name` as additional runtime parameter (the same way we've done with sycl::queue not so long ago). It's by design, so just give it a try ;)

Reply:
I've applied this comment in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/329/commits/6779a0d8bdb83c572325f602564fd286d355fcbc